projects/core/src/lib/layers/legend/legend-raster/raster-legend.component.ts

```diff
@@ -56,6 +56,52 @@ export class CastMeasurementToContinuousPipe implements PipeTransform {
     }
 }
 
+export function unifyDecimals(values: number[]): number[] {
+    // Early return if all values differ by more than 1
+    if (values.every((x) => x >= 1)) {
+        const toCompare: number[] = values.map((x) => Math.floor(x)).sort((a, b) => a - b);
+        let oneApart = true;
+        for (let i = 0; i < toCompare.length - 1; i++) {
+            if (toCompare[i + 1] - toCompare[i] < 1) {
+                oneApart = false;
+            }
+        }
+        if (oneApart) return values.map((x) => Math.floor(x));
+    }
+
+    // Find highest overlap for cut-off point
+    let maxOverlap = 0;
+    values.forEach((x) => {
+        values.forEach((y) => {
+            if (x === y) {
+                return;
+            }
+            const overlap: number = overlappingDigits(x, y);
+            maxOverlap = overlap > maxOverlap ? overlap : maxOverlap;
+        });
+    });
+    return values.map((x) => {
+        const preDecimals = Math.floor(x).toString().length;
+        const roundAt = Math.pow(10, Math.max(0, maxOverlap + 2 - preDecimals));
+        return Math.round(x * roundAt) / roundAt;
```

Comment:
What is the rationale for rounding instead of just cutting off?

Reply:
I thought it would represent the actual number more accurately. But since the goal is primarily to have distinct values, I changed it to Math.floor here (and adjusted the tests accordingly) 330864ceb13c28994b236530b8cf6744a8428b7f